File: Core/DataAccess/DataAccessLayerException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace GrowthWare.DataAccess
{
    /// <summary>
    /// Created to distinguish errors created in the data access layer.
    /// </summary>
    /// <remarks></remarks>
    [Serializable()]
    public class DataAccessLayerException : Exception
    {
        /// <summary>
        /// Creates a new instance of the DataAccessLayerException class
        /// </summary>
        public DataAccessLayerException() { }

        /// <summary>
        /// Calls base method
        /// </summary>
        /// <param name="message">string</param>
        public DataAccessLayerException(string message)
            : base(message)
        {

        }

        /// <summary>
        /// Calls base method
        /// </summary>
        /// <param name="message">string</param>
        /// <param name="innerException">Exception</param>
        public DataAccessLayerException(string message, Exception innerException)
            : base(message, innerException)
        {

        }
    }
}

File: Net_Framework/VB/GrowthWare.Framework.BusinessData/DataAccessLayer/DataAccessLayerException.vb
﻿Imports System.Runtime.Serialization

Namespace DataAccessLayer
    ''' <summary>
    ''' Created to distinguish errors created in the data access layer.
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable()>
    Public NotInheritable Class DataAccessLayerException
        Inherits Exception

        ''' <summary>
        ''' Creates a new instance of the DataAccessLayerException class
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="message">String</param>
        ''' <remarks></remarks>
        Public Sub New(ByVal message As String)
            MyBase.New(message)
        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="message">String</param>
        ''' <param name="innerException">Exception</param>
        ''' <remarks></remarks>
        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)
        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="info"></param>
        ''' <param name="context"></param>
        ''' <remarks></remarks>
        Private Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
        End Sub
    End Class

End Namespace
